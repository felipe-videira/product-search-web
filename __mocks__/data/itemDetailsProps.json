{
    "item": {
        "id": "MLA865203548",
        "title": "Microsoft Xbox One S 1tb Two-controller Bundle Blanco",
        "price": {
            "currency": "ARS",
            "amount": 55599,
            "decimals": 0
        },
        "picture": "../../__mocks__/data/product_img.jpg",
        "condition": "new",
        "free_shipping": true,
        "sold_quantity": 50,
        "description": "No solo vas a poder descargar los mejores videojuegos y navegar en la web sin límite, sino que, gracias a su interconectividad global, también vas a tener la posibilidad de competir en línea con tus amigos y con personas de todas partes del mundo disfrutando de aventuras inolvidables.\n\nCalidad de otro nivel\nSe considera que esta consola es la mejor dentro del mercado, dado que presenta una resolución de hasta 4K. Su conexión wifi te permitirá reproducir música, películas y tus series favoritas a través de las aplicaciones descargables.\n\nDiseño innovador\nMicrosoft Xbox One S ofrece impresionantes gráficos, sonido de primer nivel e incluso la posibilidad de guardar todas tus partidas en la nube.Su control combina un elegante estilo y agarre en relieve para brindar una mayor comodidad.\n\nAdaptada a tus necesidades\nGuardá tus apps, fotos, videos y mucho más en el disco duro, que cuenta con una capacidad de 1 TB. Al contar con un procesador de 8 núcleos y un procesador gráfico, brinda una experiencia dinámica, respuestas ágiles, así como transiciones fluidas de imágenes en alta definición. Por otro lado, tiene puerto USB y salida HDMI, que permiten conectar accesorios y cargar la batería de tu control mientras jugás."
    },
    "categories": [
        "Consolas y Videojuegos",
        "Consolas"
    ]
}