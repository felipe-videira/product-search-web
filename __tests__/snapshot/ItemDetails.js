import React from 'react';
import renderer from 'react-test-renderer';
import ItemDetails from '../../pages/items/[id]';
import itemDetailsProps from '../../__mocks__/data/itemDetailsProps.json';

describe('ItemDetails', () => {
  it('Should render without crashing', () => {
    const tree = renderer.create(<ItemDetails {...itemDetailsProps} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
