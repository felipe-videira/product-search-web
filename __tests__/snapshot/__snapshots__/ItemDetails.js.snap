// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemDetails Should render without crashing 1`] = `
<div>
  <nav>
    <a
      href="/"
    >
      <img
        alt="Mercado Libre"
        height="50"
        src="/assets/images/logo.png"
        width="75"
      />
    </a>
    <form
      action="/items"
      method="GET"
      name="searchBar"
      onSubmit={[Function]}
    >
      <input
        name="search"
        onChange={[Function]}
        placeholder="Nunca dejes de buscar"
        type="text"
        value={null}
      />
      <button
        type="submit"
      >
        <img
          alt="Busca"
          height="20"
          src="/assets/images/search.svg"
          width="20"
        />
      </button>
    </form>
  </nav>
  <main>
    <ul>
      <li>
        Consolas y Videojuegos
      </li>
      <li>
        Consolas
      </li>
    </ul>
    <div>
      <div>
        <div
          style={
            Object {
              "backgroundImage": "url(../../__mocks__/data/product_img.jpg)",
            }
          }
        />
        <div>
          <p>
            Nuevo - 50 vendidos
          </p>
          <h1>
            Microsoft Xbox One S 1tb Two-controller Bundle Blanco
          </h1>
          <p
            data-end="00"
          >
            $ 55,599
          </p>
          <button
            type="button"
          >
            Comprar
          </button>
        </div>
        <div>
          <h2>
            Descripción del producto
          </h2>
          <p>
            No solo vas a poder descargar los mejores videojuegos y navegar en la web sin límite, sino que, gracias a su interconectividad global, también vas a tener la posibilidad de competir en línea con tus amigos y con personas de todas partes del mundo disfrutando de aventuras inolvidables.

Calidad de otro nivel
Se considera que esta consola es la mejor dentro del mercado, dado que presenta una resolución de hasta 4K. Su conexión wifi te permitirá reproducir música, películas y tus series favoritas a través de las aplicaciones descargables.

Diseño innovador
Microsoft Xbox One S ofrece impresionantes gráficos, sonido de primer nivel e incluso la posibilidad de guardar todas tus partidas en la nube.Su control combina un elegante estilo y agarre en relieve para brindar una mayor comodidad.

Adaptada a tus necesidades
Guardá tus apps, fotos, videos y mucho más en el disco duro, que cuenta con una capacidad de 1 TB. Al contar con un procesador de 8 núcleos y un procesador gráfico, brinda una experiencia dinámica, respuestas ágiles, así como transiciones fluidas de imágenes en alta definición. Por otro lado, tiene puerto USB y salida HDMI, que permiten conectar accesorios y cargar la batería de tu control mientras jugás.
          </p>
        </div>
      </div>
    </div>
  </main>
</div>
`;
